FindANSAK.cmake

```diff
@@ -105,317 +108,357 @@ For DevStudio Generators, the picture is slightly murkier. See following comment
+            endif()
+            message( STATUS "ANSAK component location in doubt, libraries are not in a ./lib, ./lib/Debug, ./lib/<arch>/Debug Directory" )
+            message( STATUS "using ${_ansak_root} to search for include-headers" )
         endif()
     endif()
     if (_vs_match)
-        set(ANSAK_STRING_LIB "${_libs_grandparent_dirname}/$<CONFIG>/${_srch_string_lib}" CACHE FILEPATH "ANSAK String Library")
+        set( ANSAK_STRING_LIB "${_locator_parent_dir}/$<CONFIG>/ansakString.lib" CACHE FILEPATH "ANSAK String Library" )
     else()
-        set(ANSAK_STRING_LIB "${_platform_string_lib}" CACHE FILEPATH "ANSAK String Library")
-    endif()
-else()
-    # Case I, regardless of platform, if not './lib', mention that
-    set(_split_libs 0)
-    # traverse up level
-    get_filename_component(_ansak_root "${_ansak_root}" DIRECTORY)
-    if(NOT _locator_dirname STREQUAL "lib")
-        message(STATUS "ANSAK component location may be wrong: libraries are not in a ./lib directory: ${_platform_string_lib}")
-        message(STATUS "using ${_ansak_root} to search for include-headers")
+        set( ANSAK_STRING_LIB "${_locator}" CACHE FILEPATH "ANSAK String Library" )
     endif()
-    set(ANSAK_STRING_LIB "${_platform_string_lib}" CACHE FILEPATH "ANSAK String Library")
 endif()
 # _ansak_root points to ${ANSAK_DIR}, $ENV{ANSAK_DIR}, /usr, /usr/local on Linux
 # _ansak_root points to ${ANSAK_DIR}, $ENV{ANSAK_DIR}, ${PROGRAM_DATA}, $ENV{PROGRAM_DATA} or C:/ProgramData on Windows
 #      .... single value wherever it started
 # ANSAK_STRING_LIB points to libansakString.a / ansakString.lib in its directory
-# _ansak_locator_dir points to ${_ansak_root}/lib on Linux
-# _ansak_locator_dir points to whatever dir ANSAK_STRING_LIB is in, probably
+# _locator_dir points to ${_ansak_root}/lib on Linux
+# _locator_dir points to whatever dir ANSAK_STRING_LIB is in, probably
 #      ${_ansak_root}/lib on Linux, some Windows
 #      ${_ansak_root}/lib/Debug, ${_ansak_root}/lib/${_arch tag}/Debug on Windows
 #      or some other variant
 ############################################################################
 
+if( ANSAK_DEBUG )
+    message( "location development done, _ansak_root='${_ansak_root}'" )
+endif()
+
 ############################################################################
 # how many "ansak" components to we have?
 #     string.hxx            --> ansak-string
 #     runtime_exception.hxx --> ansak-lib (with or without SQLite3
 #     sqlite_exception.hxx  --> ansak-lib-sqlite3 (must Sqlite3 must be found)
 
 # find signal include files
-set(_include_dir "${_ansak_root}/include/ansak")
-find_file(ANSAK_STRING_HEADER "string.hxx" HINTS "${_include_dir}")
-find_file(ANSAK_LIB_HEADER "runtime_exception.hxx" HINTS "${_include_dir}")
-find_file(ANSAK_SQLITE_HEADER "sqlite_exception.hxx" HINTS "${_include_dir}")
+set( _include_dir "${_ansak_root}/include/ansak" )
```

Comment:
Once we've found ansak-string components, is anything else there, ready to be used.